arm: aarch32: dsb after writing to SCTLR on MPU enable/disable

manuargue opened the pull request:
This patch improves the sync barriers on Arm Aarch32 MPU enable/disable functions: move data and instructions sync barriers after writing to `SCTLR` to ensure the registers are set before proceeding and that the new changes are seen by the instructions that follow.
Tested with `fvp_baser_aemv8r_aarch32`, `qemu_cortex_r5` and in custom board.

manuargue-nxp committed:
arm: mpu: dsb after writing to SCTLR on MPU enable

Execute data and instruction sync barriers after writing to SCTLR
to enable the MPU, to ensure the registers are set before
proceeding and that the new changes are seen by the instructions
that follow.

Signed-off-by: Manuel Arguelles <manuel.arguelles@nxp.com>

stephanosio commented on arch/arm/core/aarch32/mpu/arm_mpu.c:
```diff
@@ -161,11 +162,15 @@ void arm_core_mpu_disable(void)
 	uint32_t val;
 
+	/* Force any outstanding transfers to complete before disabling MPU */
+	__DMB();
```
should this be `dsb` since `dmb` in itself does not ensure the completion of "any outstanding transfers"?

manuargue replied:
You are right, in armv8-r `dsb` ensures the completion of the memory accesses. I was misguided looking into the armv8-m [implementation](https://github.com/zephyrproject-rtos/zephyr/commit/828ae6b8bc7169d13a3c9a04ebb6c124fa2bbd45) and the fact that armv8-r also has PMSAv8 (PMSAv8-32).

stephanosio reviewed (changes requested):
Please split the changes to the enable and disable functions into two commits.

manuargue-nxp committed:
arm: mpu: dsb after writing to SCTLR on MPU disable

Execute data and instruction sync barriers after writing to SCTLR
to disable the MPU, to ensure the registers are set before
proceeding and that the new changes are seen by the instructions
that follow.

Signed-off-by: Manuel Arguelles <manuel.arguelles@nxp.com>

manuargue commented:
> Please split the changes to the enable and disable functions into two commits.

I've separated changes into two commits as requested
